tools/twix/src/panels/image_color_select.rs

```diff
@@ -0,0 +1,282 @@
+};
+
+use super::image::cycler_selector::VisionCyclerSelector;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Copy)]
+
+enum ImageKind {
+    YCbCr422,
+}
+
+pub struct ImageColorSelectPanel {
+    nao: Arc<Nao>,
+    image_buffer: ValueBuffer,
+    cycler_selector: VisionCyclerSelector,
+    brush_size: f32,
+}
+struct PixelColor {
+    red: f32,
+    green: f32,
+    blue: f32,
+}
+
+impl Panel for ImageColorSelectPanel {
+    const NAME: &'static str = "Image Color Select";
+
+    fn new(nao: Arc<Nao>, value: Option<&Value>) -> Self {
+        let cycler = value
+            .and_then(|value| value.get("cycler"))
+            .and_then(|value| value.as_str())
+            .map(Cycler::from_str)
+            .and_then(|cycler| match cycler {
+                Ok(cycler @ (Cycler::VisionTop | Cycler::VisionBottom)) => Some(cycler),
+                Ok(cycler) => {
+                    error!("Invalid vision cycler: {cycler}");
+                    None
+                }
+                Err(error) => {
+                    error!("{error}");
+                    None
+                }
+            })
+            .unwrap_or(Cycler::VisionTop);
+        let output = CyclerOutput {
+            cycler,
+            output: Output::Main {
+                path: "image".to_string(),
+            },
+        };
+        let image_buffer = nao.subscribe_output(output);
+        let cycler_selector = VisionCyclerSelector::new(cycler);
+
+        let brush_size = 50.0;
+        Self {
+            nao,
+            image_buffer,
+            cycler_selector,
+            brush_size,
+        }
+    }
+
+    fn save(&self) -> Value {
+        let cycler = self.cycler_selector.selected_cycler();
+        json!({
+            "cycler": cycler.to_string(),
+        })
+    }
+}
+
+impl Widget for &mut ImageColorSelectPanel {
+    fn ui(self, ui: &mut Ui) -> Response {
+        ui.horizontal(|ui| {
+            if self.cycler_selector.ui(ui).changed() {
+                let output = CyclerOutput {
+                    cycler: self.cycler_selector.selected_cycler(),
+                    output: Output::Main {
+                        path: "image".to_string(),
+                    },
+                };
+                self.image_buffer = self.nao.subscribe_output(output);
+            }
+            ui.add(egui::Slider::new(&mut self.brush_size, 1.0..=200.0).text("Brush"));
+        });
+
+        ui.separator();
+
+        let image = match self.get_image() {
+            Ok(image) => image,
+            Err(error) => {
+                return ui.label(format!("{error:#?}"));
+            }
+        };
+
+        let handle = ui
+            .ctx()
+            .load_texture("image", image.clone(), TextureOptions::default())
+            .id();
+        let texture = SizedTexture {
+            id: handle,
+            size: Vec2::new(image.width() as f32, image.height() as f32),
+        };
+        let response = ui.image(texture);
```

Comment:
This seems to result in the image begin of constant size. It is not scaling with the UI. I expect the image to fill the entire available space, similar to the `Image` panel